File: app/src/main/java/io/nanodegree/andrea/popularmovies/model/Movie.java
package io.nanodegree.andrea.popularmovies.model;

import com.squareup.moshi.Json;

import java.io.Serializable;

/**
 * Created by Andrea Loddo (andrea@evenly.io) on 11.05.2018
 * for Evenly GmbH
 * <p>
 * Copyright (c) 2018 Evenly GmbH,
 * all rights reserved
 */
public class Movie implements Serializable {

    private static final String MOVIES_DB_BASE_THUMBNAIL_URL = "http:/image.tmdb.org/t/p/w500";

    @Json(name = "id")
    public String id;

    @Json(name = "title")
    public String originalTitle;

    @Json(name = "poster_path")
    private String imageThumbnailUrl;

    @Json(name = "overview")
    public String plotSynopsis;

    @Json(name = "vote_average")
    public String userRating;

    @Json(name = "release_date")
    public String releaseDate;

    public String getImageThumbnailUrl() {
        return MOVIES_DB_BASE_THUMBNAIL_URL + imageThumbnailUrl;
    }
}

File: feature_movie/src/main/java/java/io/nanodegree/popularmovies/feature/movie/data/model/Movie.kt
package java.io.nanodegree.popularmovies.feature.movie.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Created by Andrea Loddo (andrea@evenly.io) on 11.05.2018
 * for Evenly GmbH
 *
 *
 * Copyright (c) 2018 Evenly GmbH,
 * all rights reserved
 */
data class Movie(@SerializedName("id") val id: String?,
                 @SerializedName("title") val originalTitle: String?,
                 @SerializedName("overview") val plotSynopsis: String?,
                 @SerializedName("vote_average") val userRating: String?,
                 @SerializedName("release_date") val releaseDate: String?,
                 @field:SerializedName("poster_path") val imageThumbnailUrl: String?) : Serializable {

    fun getFormattedImageThumbnailUrl(): String = "http:/image.tmdb.org/t/p/w500$imageThumbnailUrl"
}
